docs/latex/Inhalt/03_Inhalt/07_refactoring.tex

```diff
@@ -1,17 +1,28 @@
+Der hierfür zuständige Code ist in allen drei Methoden identisch, wurde jedoch explizit in jeder der Methoden ausgeschrieben. Hierbei handelt es sich eindeutig um Duplicated Code, einen sehr kritischen Code Smell, der dazu führen würde, dass bei einer Änderung in den Anforderungen an das \code{JPAProfileRepository} die Änderungen ggf. an drei Stellen durchgeführt werden müssten.
 
-\section{Doppelt vorhandener Code}
-Die Klasse \code{JPAProfileRepository} implementiert unter anderem die Methoden \code{persistentAdd}, \code{persistentUpdate} und \code{persistentRemove} zur Erstellung, Änderung und Löschung von Profiles. Da diese Methoden von der abstrakten Klasse \code{PersistentProfileRepository} geerbt werden, erwarten diese als Argument ein \code{PersistentProfile}. Die konkrete Implementierung \code{JPAProfileRepository} kann jedoch nur Profiles vom Typ \code{JPAProfile}, einer Unterklasse des \code{PersistentRecipe}, verarbeiten. Darum muss zu Beginn der Methoden geprüft werden, ob das übergebene Profile den passenden Typ besitzt, sonst wird eine Exception geworfen.
+\subsection{Large Class}
+Die Klasse \href{https://github.com/anditru/quickie/blob/bb41442c7f1ffbfcd3117cd86a40f7932e543a33/0-quickie-plugin-scraper/src/main/java/org/pinkcrazyunicorn/quickie/plugins/scraper/HensslerScraper.java}{\code{HensslerScraper}} hat eine stark erhöhte Zeilen- und Methodenanzahl im Vergleich zu anderen Klassen im Projekt. In dieser Klasse wird implementiert, wie Daten aus Webseite extrahiert werden und was extrahiert werden soll. Trotzdem ist hier das Single Responsibility Principle verletzt, da die Klasse sowohl implementiert, wie ein Rezept aus einer Rezeptseite extrahiert wird, aber auch wie alle Rezeptseiten gefunden werden. Aufgrund dieser Verletzung des Single Responsibility Principles und der großen Menge an Code ist die Klasse sehr unübersichtlich. So wird zum Beispiel in \href{https://github.com/anditru/quickie/blob/bb41442c7f1ffbfcd3117cd86a40f7932e543a33/0-quickie-plugin-scraper/src/main/java/org/pinkcrazyunicorn/quickie/plugins/scraper/HensslerScraper.java#L22}{Zeile 22} eine Konstante definiert, welche erst über 100 Zeilen später in der Methode \href{https://github.com/anditru/quickie/blob/bb41442c7f1ffbfcd3117cd86a40f7932e543a33/0-quickie-plugin-scraper/src/main/java/org/pinkcrazyunicorn/quickie/plugins/scraper/HensslerScraper.java#L149}{\code{getRecipeURLs}} verwendet wird. Des Weiteren verwendet diese Methode keine der anderen Methoden der Klasse und auch keine andere Methode der Klasse verwendet diese. Diese Methode hat niedrige Kohäsion in der Klasse. Dieser Code Smell sollte behoben werden, indem die Funktionalität in eine weitere Klasse ausgelagert wird.
```

Comment:
im Projekt -> des Projekts

Reply:
Trotzdem ist hier das Single Responsibility Principle verletzt -> Das "trotzdem" finde ich hier unpassend, weil der Satz davor (Es wird implememntiert WAS und WIE aus den Seiten extrahiert werden soll) auch nach einem Grund für die Verletzung des Single Responsibility Principle klingt. (Was- und Wie-Code vermischt)

Reply:
"Diese Methode hat niedrige Kohäsion in der Klasse." -> Kann eine Methode niedrige Kohäsion haben? Ich habe es so verstanden, dass das eine Eigenschaft der Klasse ist.